src/downzip-sw.js

```diff
@@ -83,7 +83,13 @@ self.addEventListener('fetch', async (event) => {
             // Append all the downloaded data
             try {
                 await new Promise((resolve, reject) => {
-                    fetch(file.downloadUrl).then(response => response.body).then(async (stream) => {
+                    fetch(file.downloadUrl).then(async response => {

```

Comment:
We still need `catch` on this `fetch`, the outer try/catch block won't catch it

Reply:
Ah right it just fails... but how come the exception in `then` block does not bubble up to the main try/catch block, if unhandled?

Reply:
`fetch` call creates a promise, `then` chains it, if an error happens in `then` block then this promise rejects. And you handle promise rejects by either setting `.catch` on it or by using `await` + `try/catch` wrapper combination.